async-query-core/src/main/java/org/opensearch/sql/spark/validator/FunctionType.java

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.validator;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+
+/**
+ * Enum for defining and looking up SQL function type based on its name. Unknown one will be
+ * considered as UDF (User Defined Function)
+ */
+@AllArgsConstructor
+public enum FunctionType {
+  AGGREGATE("Aggregate"),
+  WINDOW("Window"),
+  ARRAY("Array"),
+  MAP("Map"),
+  DATE_TIMESTAMP("Date and Timestamp"),
+  JSON("JSON"),
+  MATH("Math"),
+  STRING("String"),
+  CONDITIONAL("Conditional"),
+  BITWISE("Bitwise"),
+  CONVERSION("Conversion"),
+  PREDICATE("Predicate"),
+  CSV("CSV"),
+  MISC("Misc"),
+  GENERATOR("Generator"),
+  UDF("User Defined Function");
+
+  private final String name;
+
+  private static final Map<FunctionType, Set<String>> FUNCTION_TYPE_TO_FUNCTION_NAMES_MAP =
+      ImmutableMap.<FunctionType, Set<String>>builder()
+          .put(
+              AGGREGATE,
+              Set.of(
```

Comment:
this is spark function list? should we limit to Opensearch SQL supported function?

Reply:
Yes, it is from Spark function list. Let me check if we want to limit per function...

Reply:
should we align function with OpenSearch SQL Reference?

Reply:
The idea is that we want to allow the functions available and don't handicap the functionality.